components/wmtk_components/embedding/internal/Embedding.hpp

```diff
@@ -8,7 +8,24 @@
 #include "EmbeddingOptions.hpp"
 
 namespace wmtk::components::internal {
-
+/*
+ * @Embedding
+ * This class is used to embed a given low dimension mesh with a high dimension scaffold.
+ * The embedding component should do as least as possible, i.e. the worst resolution that is still
+ *valid. for now, this class only supports the 1-->2 dimension embedding.
+ *
+ * @input
+ * m_edges: vertices' indices pairs of the edges
+ * m_vertices: vertex position
+ * options: please see the document in EmbeddingOptions.hpp
+ * m_black_rate: used to define the Bbox size
+ * @output
+ * m_faces: 2D mesh composed of faces
+ * m_edges: 2D mesh's edges
+ * m_vertex_tags: each vertex's tag
+ * m_marked_vertices: vertices on the input edges
+ * m_marked_edges: edges on the input edges

```

Comment:
The documentation itself is fine but please write it with Doxygen-style:
https://www.doxygen.nl/manual/docblocks.html